src/typed-method-types/workflows/triggers/tests/event_test.ts

```diff
@@ -1,50 +1,166 @@
+        },
+      };
+      assertEquals(event.type, TriggerTypes.Event);
     },
-  };
-  assertEquals(event.type, TriggerTypes.Event);
-});
+  );
 
-Deno.test("Event triggers can set the type using the TriggerTypes object", () => {
-  const event: EventTrigger<ExampleWorkflow> = {
-    type: TriggerTypes.Event,
-    name: "test",
-    workflow: "#/workflows/example",
-    inputs: {},
-    event: {
-      event_type: "slack#/events/reaction_added",
-      channel_ids: ["C013ZG3K41Z"],
+  await t.step(
+    "shared_channel_invite_* event triggers do not require channel_ids",
+    () => {
+      const event: EventTrigger<ExampleWorkflow> = {
+        type: TriggerTypes.Event,
+        name: "test",
+        workflow: "#/workflows/example",
+        inputs: {},
+        event: {
+          event_type: TriggerEventTypes.SharedChannelInviteAccepted,
+        },
+      };
+      assertEquals(event.type, TriggerTypes.Event);
     },
-  };
-  assertEquals(event.type, TriggerTypes.Event);
-});
+  );
+
+  await t.step(
+    "can define a channel-scoped event with `channel_ids`",
+    () => {
+      const event: EventTrigger<ExampleWorkflow> = {
+        type: TriggerTypes.Event,
+        name: "test",
+        workflow: "#/workflows/example",
+        inputs: {},
+        event: {
+          event_type: TriggerEventTypes.ReactionAdded,
+          channel_ids: ["C1234"],
+        },
+      };
+      const _event2: EventTrigger<ExampleWorkflow> = {
+        type: TriggerTypes.Event,
+        name: "test",
+        workflow: "#/workflows/example",
+        inputs: {},
+        event: {
+          event_type: TriggerEventTypes.ReactionAdded,
+          channel_ids: ["C1234"],
+          all_resources: false,
+        },
+      };
+      assertEquals(event.type, TriggerTypes.Event);
+    },
+  );
 
-Deno.test("shared_channel_invite_* event triggers do not require channel_ids", () => {
-  const event: EventTrigger<ExampleWorkflow> = {
-    type: TriggerTypes.Event,
-    name: "test",
-    workflow: "#/workflows/example",
-    inputs: {},
-    event: {
-      event_type: "slack#/events/shared_channel_invite_accepted",
+  await t.step(
+    "can define a channel-unscoped event with `all_resources`",
+    () => {
+      const event: EventTrigger<ExampleWorkflow> = {
+        type: TriggerTypes.Event,
+        name: "test",
+        workflow: "#/workflows/example",
+        inputs: {},
+        event: {
+          event_type: TriggerEventTypes.ReactionAdded,
+          all_resources: true,
+        },
+      };
+      assertEquals(event.type, TriggerTypes.Event);
+    },
+  );
+
+  await t.step(
+    "channel event types must provide one of `all_resources:true` or `channel_ids`",
+    () => {
+      const event: EventTrigger<ExampleWorkflow> = {
+        type: TriggerTypes.Event,
+        name: "test",
+        workflow: "#/workflows/example",
+        inputs: {},
+        // @ts-expect-error requires one of `all_resources:true` or `channel_ids`
```

Comment:
Using `@ts-expect-error`, we can test for the error path, which I think is very helpful for maintaining types. If we have a large enough set of these type tests, we will be able to more easily refactor the types with less worry about forgetting something.